Websites/perf.webkit.org/tools/js/remote.js should support different "Authorization" schemes

roynos opened the pull request:
#### d7ed0ce8749d4cafc65970f662bf459b8ea43f30
<pre>
Websites/perf.webkit.org/tools/js/remote.js should support different &quot;Authorization&quot; schemes
<a href="https://bugs.webkit.org/show_bug.cgi?id=263210">https://bugs.webkit.org/show_bug.cgi?id=263210</a>
rdar://117033333

Reviewed by Dewei Zhu.

The `Authorization` header is now handled outside the `http.request` object. `server.auth` is not passed as `http.request` option.

The `NodeRemoteAPI` (aka `RemoteAPI`) now sets the `Authorization` header explicilty if `server.auth` is defined. If a scheme is specified it sets the `Authorization` header value to `&lt;scheme&gt; &lt;parameter&gt;`. Otherwise, it is assumed to be basic authentication and sets the `Authorization` header value to `Basic &lt;username&gt;:&lt;password&gt;`. This preserves backwards compatibility.

Added unit tests. `run-tests.py` will look for Mocha CLI  in another path.

* Websites/perf.webkit.org/tools/js/remote.js:
(NodeRemoteAPI.prototype.configure):
(NodeRemoteAPI.prototype.sendHttpRequest):
* Websites/perf.webkit.org/tools/run-tests.py:
(main):
* Websites/perf.webkit.org/unit-tests/tools-js-remote-tests.js:
(assert.doesNotThrow):
(assert.throws):

Canonical link: <a href="https://commits.webkit.org/269771@main">https://commits.webkit.org/269771@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/49e3ffdad4109881806c2bca86aa7d6fc7eed047

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23422 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24545 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21633 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23693 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3082 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23973 "Built successfully") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23663 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1091 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20277 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26186 "Built successfully") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21171 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27427 "4 flakes 114 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21354 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21431 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25178 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/884 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18612 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/845 "Built successfully") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/5618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1294 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1149 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dewei-zhu commented on Websites/perf.webkit.org/tools/js/remote.js:
```diff
@@ -108,6 +111,10 @@ class NodeRemoteAPI extends CommonRemoteAPI {
             if (this._cookies.size)
                 request.setHeader('Cookie', Array.from(this._cookies.keys()).map((key) => `${key}=${this._cookies.get(key)}`).join('; '));
 
+            if (server.auth) {
```
nit: no `{}` for oneline statement

roynos replied:
fixed.

dewei-zhu reviewed (approved):
r=me

roynos committed:
Websites/perf.webkit.org/tools/js/remote.js should support different "Authorization" schemes
https://bugs.webkit.org/show_bug.cgi?id=263210
rdar://117033333

Reviewed by Dewei Zhu.

The `Authorization` header is now handled outside the `http.request` object. `server.auth` is not passed as `http.request` option.

The `NodeRemoteAPI` (aka `RemoteAPI`) now sets the `Authorization` header explicilty if `server.auth` is defined. If a scheme is specified it sets the `Authorization` header value to `<scheme> <parameter>`. Otherwise, it is assumed to be basic authentication and sets the `Authorization` header value to `Basic <username>:<password>`. This preserves backwards compatibility.

Added unit tests. `run-tests.py` will look for Mocha CLI  in another path.

* Websites/perf.webkit.org/tools/js/remote.js:
(NodeRemoteAPI.prototype.configure):
(NodeRemoteAPI.prototype.sendHttpRequest):
* Websites/perf.webkit.org/tools/run-tests.py:
(main):
* Websites/perf.webkit.org/unit-tests/tools-js-remote-tests.js:
(assert.doesNotThrow):
(assert.throws):

Canonical link: https://commits.webkit.org/269771@main
